pmd-apex/src/main/resources/category/apex/bestpractices.xml

```diff
@@ -172,6 +172,68 @@ public class Foo {
         </example>
     </rule>
 
+    <rule name="AvoidFutureAnnotation"
+          since="7.19.0"
+          language="apex"
+          message="Usage of @Future annotation should be limited. Consider implementing the Queueable interface instead."
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_bestpractices.html#avoidfutureannotation">
+        <description>
+Usage of the `@Future` annotation should be limited. The `@Future` annotation is a legacy way to execute asynchronous Apex code, but it has several limitations:
+- Methods must be static
+- Only primitive data types, primitive arrays, or collections of primitive data types can be parameters
+- No way to monitor job status or handle failures
+- No support for chaining jobs
+
+Consider implementing the `Queueable` interface instead, which provides:
+- Better error handling and monitoring capabilities
+- Support for more complex data types
+- Ability to chain jobs
+
+See more here: &lt;https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_queueing_jobs.htm&gt;
+        </description>
```

Comment:
Should we also mention the rule [QueueableWithoutFinalizer](https://docs.pmd-code.org/latest/pmd_rules_apex_bestpractices.html#queueablewithoutfinalizer) or at least include the finalizer in the example code?